prog/weaver/main.go

```diff
@@ -441,6 +442,7 @@ func main() {
 		HandleHTTP(muxRouter, version, router, allocator, defaultSubnet, ns, dnsserver, &waitReady)
 		HandleHTTPPeer(muxRouter, allocator, discoveryEndpoint, token, name.String())
 		muxRouter.Methods("GET").Path("/metrics").Handler(metricsHandler(router, allocator, ns, dnsserver))
+		muxRouter.Methods("GET").Path("/proxyaddrs").HandlerFunc(proxy.StatusHTTP)
```

Comment:
If `!proxyConfig.Enabled`, `proxy` will be `nil`.
Will not this panic with `panic: runtime error: invalid memory address or nil pointer dereference`?